code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/deployment/ServiceDeploymentPlanner.java

```diff
@@ -28,7 +28,7 @@ public ServiceDeploymentPlanner(List<Service> services, List<DeploymentUnit> uni
 
         if (units != null) {
             for (DeploymentUnit unit : units) {
-                if (unit.isUnhealthy()) {
+                if (unit.isUnhealthy() || !unit.isHostActive()) {
```

Comment:
I see your point. What if we leave isHostActive in deploymentUnit, but make it private, and call from within unit.isUnhealthy. Then it would be easy to separate it in the future if there is a need for it.